tron/serialize/runstate/statemanager.py

```diff
@@ -160,16 +173,20 @@ def restore(self, job_names, skip_validation=False):
         return state
 
     def _restore_runs_for_job(self, job_name, job_state):
+        """Restore the state for the runs of each job"""
         run_nums = job_state["run_nums"]
-        runs = []
-        for run_num in run_nums:
-            key = jobrun.get_job_run_id(job_name, run_num)
-            run_state = list(self._restore_dicts(runstate.JOB_RUN_STATE, [key]).values())
-            if not run_state:
-                log.error(f"Failed to restore {key}, no state found for it")
-            else:
-                runs.append(run_state[0])
-        return runs
+        keys = [jobrun.get_job_run_id(job_name, run_num) for run_num in run_nums]
+        job_runs_restored_states = self._restore_dicts(runstate.JOB_RUN_STATE, keys)
+        run_state = copy.copy(job_runs_restored_states)
+        for key, value in run_state.items():
+            if value == {}:
+                log.error(f"Failed to restore {key}, no state found for it!")
+                job_runs_restored_states.pop(key)
```

Comment:
```suggestion
        for run_id, state in run_state.items():
            if state == {}:
                log.error(f"Failed to restore {run_id}, no state found for it!")
                job_runs_restored_states.pop(run_id)
```
might be a little more self-documenting (although, you'll want to verify that i'm reading this right and that these are indeed valid renames :p)